From 76d32595e43a08df299101467d162de2391f3482 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=93=8D=E9=A9=AC?= <lion@9465.net>
Date: Sun, 18 Feb 2024 06:57:57 +0800
Subject: [PATCH 021/228] chachap10-ppc.pl: Fix truncated relocation

Fix error: relocation truncated to fit: R_PPC64_REL14 (stub)
against symbol `ChaCha20_ctr32_vsx_8x'

CLA: trivial

Reviewed-by: Tom Cosgrove <tom.cosgrove@arm.com>
Reviewed-by: Tomas Mraz <tomas@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/23618)

(cherry picked from commit 1afb326da4b3a781dc0d7cd91262d81126783b07)
---
 crypto/chacha/asm/chachap10-ppc.pl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/crypto/chacha/asm/chachap10-ppc.pl b/crypto/chacha/asm/chachap10-ppc.pl
index 78cd4aed0d..54a1dd4590 100755
--- a/crypto/chacha/asm/chachap10-ppc.pl
+++ b/crypto/chacha/asm/chachap10-ppc.pl
@@ -170,7 +170,9 @@ $code.=<<___;
 .align	5
 .ChaCha20_ctr32_vsx_p10:
 	${UCMP}i $len,255
-	bgt 	ChaCha20_ctr32_vsx_8x
+	ble	.Not_greater_than_8x
+	b	ChaCha20_ctr32_vsx_8x
+.Not_greater_than_8x:
 	$STU	$sp,-$FRAME($sp)
 	mflr	r0
 	li	r10,`15+$LOCALS+64`
-- 
2.39.5 (Apple Git-154)

